### PR TITLE
Fix boxZoom._moved remaining in bad state after use (cf #3026)

### PR DESCRIPTION
**Pull request by yohanboniface**
This very small fix seems to fix it for me, both on Firefox and Chromium.
I can't see a reason for this not to work on other browser, but I may have missed some edge case to test.

### CODE REVIEW

**Comment by yohanboniface** [comment]
Found an edge case: hitting `escape` while drawing the box to cancel the action doesn't reset `_moved` state.

**Comment by mourner** [comment]
Also, as far as I remember, the whole thing about remaining `_moved` state was about not firing the click event after something like box zoom. Needs checking.

**Comment by yohanboniface** [comment]
>  the whole thing about remaining _moved state was about not firing the click event after something like box zoom

Good point, seems less trivial then!

**Comment by mourner** [comment]
Maybe resetting it on something like setTimeout 0 would suffice, since click/mouseup/dragend should all happen in the same JS tick.

**Comment by yohanboniface** [comment]
> Maybe resetting it on something like setTimeout 0 would suffice

Works here both in FF and Chromium, in the normal boxZoom scenario plus hitting ESC while drawing the box.

For the record, FF do not issue a `click` event at all after a drag, while Chromium do.

Anyone to test it on other browsers/OS? On the master, just open the `debug/map/map.html` page in the browser, and check that `click` isn't logged into the console after a boxZoom nor a boxZoom + ESC (while doing a basic click on the map should log `click`.

**Comment by mourner** [comment]
> For the record, FF do not issue a click event at all after a drag, while Chromium do.

Even after the setTimeout fix?

File: src/map/handler/Map.BoxZoom.js [hunk 1]
```diff
@@ -26,6 +26,17 @@ L.Map.BoxZoom = L.Handler.extend({
...
 
+	abort: function () {
+		this._resetState();
+		this._finish();
+	},
+
+	_resetState: function () {
+		// Postpone to next JS tick so internal click event handling
+		// still see it as "moved".
+		setTimeout(L.bind(this._resetState, this), 0);
```
Comment (mourner) [inline src/map/handler/Map.BoxZoom.js]: this looks weird, isn't it going to an infinite loop? where is `this._moved = false` part?

Comment (yohanboniface) [inline src/map/handler/Map.BoxZoom.js]: Meeeeh, last time refactor + without unittests => BAM ;)

**Comment by yohanboniface** [comment]
> Even after the setTimeout fix?

No, I meant before the fix, sorry for not being clear.
